### DISCUSSION TITLE
rotor compresser R37 modification

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#662] [General]
hello profressor PingHe, I would like to know what can I do when the optmization objective(Torque) reduce very little(0.24%). I modify from rortor37 to rotor67, using same objective and constaint. which parameter is good for making the toque reduce to more lower value. Is it because residual is not low enough my case all residual is below 1e-5 but I see in the profressorr tutorial profressor set it as 1E-8. I have try to set fvSolution and fvScheme but I can not get as low as profressor set up. Eventhough my case can not get to below 1E-8 but shock wave can be captured. so I really wonder how to make it reduce. For this case how much reduction can be treated as good and I have make a validation the total pressure ratio error is 10% since we have no rotor tip is it ok for 100k Mesh. because I have try 100k 300k 400k 500k and the more higher resolution like 300k 400k 500k can get the presure residual down to just only 1E-4 which is not good enough for capturing the shock wave.
![image](https://github.com/mdolab/dafoam/assets/115110434/bc43bf97-e3f8-4dfa-94e1-3d8b92ae4fa1) 
I have another question I see profressor set niuTiuda very high is it help in convergence. If set it like this will solution will be realistic since normal flow condition is around 1e-5 so I am curious
![image](https://github.com/mdolab/dafoam/assets/115110434/3e3fd502-9fe9-4924-a5cb-766f431c6693)

**Comment by friedenhe** [comment]
(1) Did the adjoint converge? (2) I suggest you try a different optimizer, like SLSQP. (3) In the 2nd screenshot, nuTilda=1e-3 is for setting normalizing value for adjoint, it is not used in flow simulations.

**Reply by Ryupuriwatt** [reply]
For aerodynamic optimization adjoint converge I can reduce torque 1.1 percent but in the aerostructural optimization torque reduce just around 0.4%. I am not sure the reason that make the objective reduce very slightly.

**Comment by friedenhe** [comment]
Have you tried SLSQP? IPOPT sometimes has problem converging constrained shape optimization.

**Reply by Ryupuriwatt** [reply]
![image](https://github.com/user-attachments/assets/b0b99eb0-b1a3-46fc-865c-8b4a5d521e66)
Ipopt have this problem I still trying to find a way to use it

**Comment by friedenhe** [comment]
Are you using v2 scripts? This is a known issue (SLSQP does not work for v2 scripts...), and we haven't gotten a chance to fix it. Please use a v3 script.

**Reply by Ryupuriwatt** [reply]
Profressor this is the script that I used for running aero-structural optimization of rotor67. IPOPT works fine but due to the constraint problem I can not get a good result and now I try to use SLSQP instead but there is a problem when run it even though I think this script is version3 right because aerostructural optimization used OPENMDAO framework. And I also test your latest tutorial S809, MACH_tutorial_wing using v3 script I also get an error. could processor give me an suggustion or any thing in my script to be able to run it.

**Comment by friedenhe** [comment]
OK, you are using v3 script.. Then, the SLSQP bug is strange. Can you run the NACA0012 tutorial case using SLSQP optimizer and see if you can reproduce the SLSQP error you saw?

**Reply by Ryupuriwatt** [reply]
Now I have fix this by modify some part of the code thank you very much profressor PingHe.
